2 new failures in `maint`  on datalad/git-annex

A sattelite to #5759 : two other failures creeped in  while those prior 2 regressions (?) were not attended to

```
ERROR: datalad.support.tests.test_external_versions.test_external_versions_basic
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/opt/hostedtoolcache/Python/3.7.10/x64/lib/python3.7/site-packages/nose/case.py", line 198, in runTest
    self.test(*self.arg)
  File "/opt/hostedtoolcache/Python/3.7.10/x64/lib/python3.7/site-packages/datalad/support/tests/test_external_versions.py", line 57, in test_external_versions_basic
    assert_greater(ev[our_module], '0.1')
  File "/opt/hostedtoolcache/Python/3.7.10/x64/lib/python3.7/unittest/case.py", line 1249, in assertGreater
    if not a > b:
  File "/opt/hostedtoolcache/Python/3.7.10/x64/lib/python3.7/distutils/version.py", line 64, in __gt__
    c = self._cmp(other)
  File "/opt/hostedtoolcache/Python/3.7.10/x64/lib/python3.7/distutils/version.py", line 337, in _cmp
    if self.version < other.version:
TypeError: '<' not supported between instances of 'str' and 'int'
```

super fresh
```
$> git grep 'test_external_versions_basic.*ERR'
cron-20210707/build-macos.yaml-344-ac85ac50-failed/2_test-datalad (maint).txt:2021-07-07T04:07:10.3761420Z datalad.support.tests.test_external_versions.test_external_versions_basic ... ERROR
cron-20210707/build-macos.yaml-344-ac85ac50-failed/test-datalad (maint)/11_Run datalad tests.txt:2021-07-07T04:07:10.3761400Z datalad.support.tests.test_external_versions.test_external_versions_basic ... ERROR
cron-20210707/build-ubuntu.yaml-336-ac85ac50-failed/2_test-datalad (maint).txt:2021-07-07T04:02:34.5914734Z datalad.support.tests.test_external_versions.test_external_versions_basic ... ERROR
cron-20210707/build-ubuntu.yaml-336-ac85ac50-failed/test-datalad (maint)/12_Run datalad tests.txt:2021-07-07T04:02:34.5914720Z datalad.support.tests.test_external_versions.test_external_versions_basic ... ERROR
cron-20210707/build-windows.yaml-307-ac85ac50-failed/2_test-datalad (maint).txt:2021-07-07T05:37:52.8466519Z datalad.support.tests.test_external_versions.test_external_versions_basic ... ERROR
cron-20210707/build-windows.yaml-307-ac85ac50-failed/test-datalad (maint)/21_Run datalad tests.txt:2021-07-07T05:37:52.8466481Z datalad.support.tests.test_external_versions.test_external_versions_basic ... ERROR
```

# 2

```
======================================================================
FAIL: datalad.cmdline.tests.test_main.test_script_shims
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/opt/hostedtoolcache/Python/3.7.10/x64/lib/python3.7/site-packages/nose/case.py", line 198, in runTest
    self.test(*self.arg)
  File "/opt/hostedtoolcache/Python/3.7.10/x64/lib/python3.7/site-packages/datalad/cmdline/tests/test_main.py", line 223, in test_script_shims
    assert get_numeric_portion(version) # that my lambda is correctish
AssertionError
```
also seems to be super-fresh:
```
(git)smaug:/mnt/btrfs/datasets/datalad/ci/git-annex/builds/2021/07[master]git
$> git grep 'test_script_shims.*FAIL'
cron-20210707/build-macos.yaml-344-ac85ac50-failed/2_test-datalad (maint).txt:2021-07-07T02:31:15.6066460Z datalad.cmdline.tests.test_main.test_script_shims ... FAIL
cron-20210707/build-macos.yaml-344-ac85ac50-failed/test-datalad (maint)/11_Run datalad tests.txt:2021-07-07T02:31:15.6066410Z datalad.cmdline.tests.test_main.test_script_shims ... FAIL
cron-20210707/build-ubuntu.yaml-336-ac85ac50-failed/2_test-datalad (maint).txt:2021-07-07T02:54:21.8669030Z datalad.cmdline.tests.test_main.test_script_shims ... FAIL
cron-20210707/build-ubuntu.yaml-336-ac85ac50-failed/test-datalad (maint)/12_Run datalad tests.txt:2021-07-07T02:54:21.8668990Z datalad.cmdline.tests.test_main.test_script_shims ... FAIL
cron-20210707/build-windows.yaml-307-ac85ac50-failed/2_test-datalad (maint).txt:2021-07-07T03:55:15.6530684Z datalad.cmdline.tests.test_main.test_script_shims ... FAIL
cron-20210707/build-windows.yaml-307-ac85ac50-failed/test-datalad (maint)/21_Run datalad tests.txt:2021-07-07T03:55:15.6530636Z datalad.cmdline.tests.test_main.test_script_shims ... FAIL
```

so both likely to relate somehow to versioneer and new releasing setup.  FWIW the `#2` fails differently for me locally on maint
```
======================================================================
FAIL: datalad.cmdline.tests.test_main.test_script_shims
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/tmp/venvs/dev3/lib/python3.9/site-packages/nose/case.py", line 198, in runTest
    self.test(*self.arg)
  File "/home/yoh/proj/datalad/datalad-maint/datalad/cmdline/tests/test_main.py", line 224, in test_script_shims
    assert_equal(get_numeric_portion(__version__),
AssertionError: Lists differ: ['0', '14'] != ['0', '14', '6']

Second list contains 1 additional elements.
First extra element 2:
'6'

- ['0', '14']
+ ['0', '14', '6']
?           +++++
```
will submit PR on this one now...

--- a/datalad/__init__.py
+++ b/datalad/__init__.py
@@ -263,6 +263,17 @@ def teardown_package():
     AnnexRepo._ALLOW_LOCAL_URLS = False  # stay safe!
 
 
-lgr.log(5, "Done importing main __init__")
-
 from .version import __version__
+
+if str(__version__) == '0' or __version__.startswith('0+'):
+    lgr.warning(
+        "DataLad was not installed 'properly' so its version is an uninformative %r.\n"
+        "It can happen e.g. if datalad was installed via\n"
+        "  pip install https://github.com/.../archive/{commitish}.zip\n"
+        "instead of\n"
+        "  pip install git+https://github.com/...@{commitish} .\n"
+        "We advise to re-install datalad or downstream projects might not operate correctly.",
+        __version__
+    )
+
+lgr.log(5, "Done importing main __init__")

